nemo/core/config/optimizers.py

```diff
@@ -238,6 +240,11 @@ def get_optimizer_config(name: str, **kwargs: Optional[Dict[str, Any]]) -> Optim
         )
 
     scheduler_params = AVAILABLE_OPTIMIZER_PARAMS[name]
+
+    if kwargs is not None and len(kwargs) != 0:
+        kwargs = OmegaConf.create(kwargs)
+        OmegaConf.merge(scheduler_params(), kwargs)
```

Comment:
I've been thinking more about this construction. We honestly could do away with dataclasses if we just use `hydra.util.instantiate` instead and pass normal python arugments as opposed to DictConfigs to our classes.

Reply:
Instantiate requires the assumption you are following the `target` and `params` pattern which causes very deep nest when trying to override parameters for optimizers - `model.optim.sched.params.weight_decay` which is untenable.

Reply:
This exposes an issue with the construction of our yaml files, not a problem with hydra.
Nothing is stopping us from exposing it at `sched.params.weight_decay`, or `sched.weight_decay`, or even `weight_decay`.

Reply:
Well we need to make certain assumptions regarding this, and using classpath everywhere is problematic in all cases other than pure productions configs (which we won't even need there since we will simply be using config objects) so for research flexibility the current schema is practical and useful enough.